### PR TITLE
Initial jetpack implementation

### PR DESCRIPTION
**Pull request by ShadowBrian**
https://github.com/apetavern/grubs/assets/11138884/2d6335ad-6855-4f21-9b03-b8dffa3f242d

### CODE REVIEW

File: code/Equipment/Tools/JetpackTool.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
+using Sandbox;
+using Sandbox.Utility;
+
+namespace Grubs.Equipment.Tools;
+
+public sealed class JetpackTool : Tool
+{
+	private float _jetpackDir { get; set; }
```
Comment (trundlr) [inline code/Equipment/Tools/JetpackTool.cs:8]: Property should always be PascalCase

File: code/Equipment/Tools/JetpackTool.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+	[Property] private float MaxJetFuel { get; set; } = 10f;
+
+	private float _currentJetFuel { get; set; }
```
Comment (trundlr) [inline code/Equipment/Tools/JetpackTool.cs:12]: Same here as above

File: code/Equipment/Tools/JetpackTool.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+	[Property] public GameObject FBFlame { get; set; }
+
+	[Property] public GameObject UDFlame1 { get; set; }
+
+	[Property] public GameObject UDFlame2 { get; set; }
+
+	[Sync] private float _fbFlameScale { get; set; }
+
+	[Sync] private float _udFlameScale { get; set; }
```
Comment (trundlr) [inline code/Equipment/Tools/JetpackTool.cs:47]: PascalCase

File: code/Equipment/Tools/JetpackTool.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		UDFlame2.Enabled = false;
+		IsFiring = false;
+		_currentJetFuel = MaxJetFuel;
+	}
+
+	[Property] public GameObject FBFlame { get; set; }
+
+	[Property] public GameObject UDFlame1 { get; set; }
+
+	[Property] public GameObject UDFlame2 { get; set; }
```
Comment (trundlr) [inline code/Equipment/Tools/JetpackTool.cs:43]: I have no clue what these properties represent. Could we name them something a little more clear?

File: code/Equipment/Tools/JetpackTool.cs [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	{
+		var characterController = Equipment.Grub.CharacterController;
+
+		FBFlame.Enabled = true;
+		UDFlame1.Enabled = true;
+		UDFlame2.Enabled = true;
+
+		Transform Middle_Flame = Equipment.Model.GetAttachment( "jet_middle" ).Value;
+		Transform Left_Flame = Equipment.Model.GetAttachment( "jet_left" ).Value;
+		Transform Right_Flame = Equipment.Model.GetAttachment( "jet_right" ).Value;
```
Comment (trundlr) [inline code/Equipment/Tools/JetpackTool.cs:59]: Use var wherever possible

File: code/Equipment/Tools/JetpackTool.cs [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+			}
+
+			animator.JetpackDir = _jetpackDir;
+		}
+	}
+
+	private void UpdateRotation()
+	{
+		var characterController = Equipment.Grub.CharacterController;
+		Rotation targetRotation = -characterController.Velocity.x switch
```
Comment (trundlr) [inline code/Equipment/Tools/JetpackTool.cs:131]: var

File: code/Equipment/Tools/JetpackTool.cs (hunk 4 above)
Comment (ShadowBrian) [reply code/Equipment/Tools/JetpackTool.cs:43]: Forward/Back flame
Up/Down flame

File: code/Equipment/Tools/JetpackTool.cs (hunk 6 above)
Comment (ShadowBrian) [reply code/Equipment/Tools/JetpackTool.cs:131]: I tried using var but the switch case was complaining

**Review by trundlr — APPROVED** [review]

**Pull request merged**